src/plugins/intel_cpu/src/utils/caseless.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @file caseless.hpp
+ * @brief A header file with caseless containers
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <cctype>
+#include <functional>
+#include <iterator>
+#include <map>
+#include <set>
+#include <unordered_map>
+
+namespace ov {
+namespace intel_cpu {
+
+/**
+ * @brief Provides caseless comparison for STL algorithms
+ *
+ * @tparam Key type, usually std::string
+ */
+template <class Key>
+class CaselessLess {
```

Comment:
@riverlijunjie could you also move src/inference/tests/functional/caseless_tests.cpp to CPU unittests?

Reply:
sure, will do it.

Reply:
done